src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -838,6 +839,15 @@ private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IEnumerable<Configuratio
 
                 try
                 {
+                    Uri currentEndpoint = _configClientManager.GetEndpointForClient(currentClient);
+
+                    if (previousEndpoint != currentEndpoint)
+                    {
+                        _logger.LogWarning(LogHelper.BuildFailoverToDifferentEndpointMessage(previousEndpoint.ToString(), currentEndpoint.ToString()));
+                    }
```

Comment:
This seems like it should be after the finally block.

Reply:
I agree from how it looks, but then it seems like I need to either make an extra call to `GetEndpointforClient` or update `currentEndpoint` at the bottom of the while loop and I wasn't sure which made more sense.